crates/apub/src/fetcher/search.rs

```diff
@@ -44,6 +44,18 @@ pub(crate) async fn search_query_to_object_id(
   })
 }
 
+/// Converts a search query to an object id.  The query MUST bbe a URL which will bbe treated
+/// as the ObjectId directly.  If the query is a webfinger identifier (@user@example.com or
+/// !community@example.com) this method will return an error.
+#[tracing::instrument(skip_all)]
+pub(crate) async fn search_query_to_object_id_local(
+  query: &str,
+  context: &Data<LemmyContext>,
+) -> Result<SearchableObjects, LemmyError> {
+  let url = Url::parse(query)?;
+  ObjectId::from(url).dereference_local(context).await
```

Comment:
You can directly call `ObjectId::parse` which calls `Url::parse` internally. And in fact it doesnt make much sense to have a separate function for this single line.